### ISSUE TITLE
Keydown, keyup, keypress pages should mention they work on document, body, etc

### ISSUE THREAD

**Issue by jeremyredhead** [mdn/content#28561]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event

### What specific section or headline is this issue about?

main section(?)

### What information was incorrect, unhelpful, or incomplete?

The keydown, keyup, and keypress pages don't explicitly mention that keyboard events are also "generated"/can be installed on by `window`, `document`, `<html>` and `<body>`. The keydown page *implies* that is the case via the sentence 
<q>If not caught, they [bubble](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#event_bubbling) up the [DOM tree](https://developer.mozilla.org/en-US/docs/Web/API/Document_object_model/Using_the_Document_Object_Model#what_is_a_dom_tree) until they reach [Document](https://developer.mozilla.org/en-US/docs/Web/API/Document)</q>, but it would be better to be explicit, and the the keyup page currently lacks this sentence. (The keypress article currently fails to also mention the preceding sentence: <q>Keyboard events are only generated by `<input>`, `<textarea>`, `<summary>` and anything with the contentEditable or tabindex attribute</q>)

### What did you expect to see?

The keydown, keyup, and keypress pages should explicitly mention that keyboard events are "generated"/installable on `window`, `document`, `<html>` and `<body>`

### Do you have anything more you want to share?

I would've edited the pages myself but I wasn't sure how to (re)write them to incorporate the info. (also, editing via github web is annoying)

Also, I suppose this should be a separate issue, but the pages also fail to mention that elements with implicit `tabindex`s (such as `<a>`) also generate keyboard events.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/element/keydown_event`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/element/keydown_event/index.md
* Last commit: https://github.com/mdn/content/commit/bbf7f25f9cf95fb154e2740a9fdc9c02818981bf
* Document last modified: 2023-04-07T05:55:26.000Z

</details>

**Comment by JayRKyd** [comment]
I would like to tackle this issue please.

**Comment by hamishwillee** [comment]
@JayRKyd That would be good. 

We no longer directly assign items. Instead you self assign by creating a PR and cross linking to the issue (by adding `Fixes #28561` in the PR that fixes this issue). Or to put it another way, before starting an issue, check that there is no PR linked to it already.

**Comment by wirekang** [comment]
Thank you for creating a great issue. I was so confused how `keydown` event listener on `document` works well without propagation.

> The keydown page implies that is the case via the sentence
If not caught, they bubble up the DOM tree until they reach Document,
but it would be better to be explicit,

I don't think the sentence implies the fact that document(or body, etc..) can generates the event itself. There's absolutely no information about it in the MDN documentation.

> The keydown, keyup, and keypress pages should explicitly mention that keyboard events are "generated"/installable on window, document, \<html\> and \<body\>

I'm wondering how did you know this information? I want to ensure this behavior is standard or guaranteed by modern browsers at least. Is there any objective evidence?

**Comment by jeremyredhead** [comment]
> > The keydown page implies that is the case via the sentence "If not caught, they bubble up the DOM tree until they reach Document", but it would be better to be explicit,
>
> I don't think the sentence implies the fact that document(or body, etc..) can generates the event itself. There's absolutely no information about it in the MDN documentation.

Perhaps 

> > The keydown, keyup, and keypress pages should explicitly mention that keyboard events are "generated"/installable on window, document, `<html>` and `<body>`
> 
> I'm wondering how did you know this information? I want to ensure this behavior is standard or guaranteed by modern browsers at least. Is there any objective evidence?

I know from experience/testing. Unfortunately, the specification ([UI Events](https://w3c.github.io/uievents/)) doesn't seem to mention `document` or `window` for any of [`keydown`][], [`keyup`][], or [`keypress`][], It is implied, I think, that `<html>` and `<body>` would naturally work, since all three list simply "Element" for their "Trusted Targets" (cf. [`input`][] and [`beforeinput`][] whose trusted targets are "Element (specifically: control types such as HTMLInputElement, etc.) or any Element with contenteditable attribute enabled" (shockingly loose language!))

Note that the browser compatibility table on the [MDN keypress article][] mentions this for Firefox: 
> As of Firefox 65, the keypress event is no longer fired for [non-printable keys](https://developer.mozilla.org/docs/Web/API/KeyboardEvent/keyCode#Non-printable_keys_(function_keys)), except for the Enter key, and the Shift + Enter and Ctrl + Enter key combinations (these were kept for cross-browser compatibility purposes). 

However nothing of this sort is mentioned in the UI Events specification, for keypress, or anywhere.  
(There is also a compatibility note for Chromium et al, but I suspect that is obviated by the text in UI Events "if and only if that key normally produces a [character value](https://w3c.github.io/uievents/#character-value)" as it seems that Ctrl+(anything but Enter) does not produce a keypress event in Firefox (even Ctrl+Shift+Enter -- despite the fact that Ctrl+Enter & Shift+Enter both do!)

So, I am not sure that an authoritative specification source for this behavior can be found. It appears to be yet another thing that browsers just "learned to do" from each other but (have still) never bothered to properly write down. Someone perhaps could go through and make an exhaustive cross-browser chart comparing "unspecified" behaviors in keydown/keyup/keypress, but that person will not be me, at least not today, as writing this reply has tired me out :P

[`keydown`]: https://w3c.github.io/uievents/#event-type-keydown
[`keypress`]: https://w3c.github.io/uievents/#event-type-keypress
[`keyup`]: https://w3c.github.io/uievents/#event-type-keyup
[`input`]: https://w3c.github.io/uievents/#input
[`beforeinput`]: https://w3c.github.io/uievents/#beforeinput
[MDN keypress article]: https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event
